plugins/modules/web_infrastructure/jira.py

```diff
@@ -246,13 +292,18 @@ def request(url, user, passwd, timeout, data=None, method=None):
     # inject the basic-auth header up-front to ensure that JIRA treats
     # the requests as authorized for this user.
     auth = to_text(base64.b64encode(to_bytes('{0}:{1}'.format(user, passwd), errors='surrogate_or_strict')))
-
     response, info = fetch_url(module, url, data=data, method=method, timeout=timeout,
                                headers={'Content-Type': 'application/json',
                                         'Authorization': "Basic %s" % auth})
 
     if info['status'] not in (200, 201, 204):
-        module.fail_json(msg=info['msg'])
+        module.fail_json(msg=info)
+        error = json.loads(info['body'])
```

Comment:
```suggestion
        error = json.loads(info['body'])
```
I think it (and below) will never be reached because of the previous line

Reply:
This is an error. I have remove it